docs/developers_guide/deploying_spack.rst

```diff
@@ -0,0 +1,482 @@
+        --mache_fork xylar/mache \
+        --mache_branch update_cime_machines \
+        --update_spack \
+        --spack /lcrc/group/e3sm/${USER}/spack_test \
+        --tmpdir ${TMPDIR} \
+        --compiler intel gnu \
+        --mpi openmpi openmpi \
+        --with_netlib_lapack \
+        --with_petsc \
+        --recreate
+
+
+Troubleshooting spack
+------------------------------
+
+If you encounter an error like:
+.. code-block:: bash
+
+    ==>   spack env activate dev_compass_1_2_0-alpha_4_gnu_mpich
+    ==> Error: Package 'armpl' not found.
+    You may need to run 'spack clean -m'.
+
+during the attempt to build spack, you will first need to find the path to `setup-env.sh` (see 
+`compass/build_*/build*.sh`) and source that script to get the `spack` command, e.g.:
+
+.. code-block:: bash
+
+    source ${PSCRATCH}/spack_test/spack_for_mache_1.12.0/share/spack/setup-env.sh
+
+Then run the suggested command:
+
+.. code-block:: bash
+
+    spack clean -m
+
+After that, re-running `./conda/configure_compass_env.py` should work correctly.
+
+This issue seems to be related to switching between spack v0.18 and v0.19 (used by different versions of compass).
+
+Testing compass
+===============
+
+Testing MPAS-Ocean without PETSc
+--------------------------------
+
+Please use the E3SM-Project submodule in compass for testing, rather than
+E3SM’s master branch.  The submodule is the version we know works with compass
+and serves as kind of a baseline for other testing.
+
+.. code-block:: bash
+
+    # source whichever load script is appropriate
+    source load_dev_compass_1.2.0-alpha.5_chrysalis_intel_openmpi.sh
+    git submodule update --init --recursive
+    cd E3SM-Project/components/mpas-ocean
+    # this will build with scorpio and openmp
+    make ifort
+    compass suite -s -c ocean -t pr -p . \
+        -w /lcrc/group/e3sm/ac.xylar/compass/test_20230202/ocean_pr_chrys_intel_openmpi
+    cd /lcrc/group/e3sm/ac.xylar/compass/test_20230202/ocean_pr_chrys_intel_openmpi
+    sbatch job_script.pr.bash
+
+You can make other worktrees of E3SM-Project for testing other compilers if
+that’s helpful.  It also might be good to open a fresh terminal to source a
+new load script.  This isn’t required but you’ll get some warnings.
+
+.. code-block:: bash
+
+    source load_dev_compass_1.2.0-alpha.5_chrysalis_gnu_openmpi.sh
+    cd E3SM-Project
+    git worktree add ../e3sm_chrys_gnu_openmpi
+    cd ../e3sm_chrys_gnu_openmpi
+    git submodule update --init --recursive
+    cd components/mpas-ocean
+    make gfortran
+    compass suite -s -c ocean -t pr -p . \
+        -w /lcrc/group/e3sm/ac.xylar/compass/test_20230202/ocean_pr_chrys_gnu_openmpi
+    cd /lcrc/group/e3sm/ac.xylar/compass/test_20230202/ocean_pr_chrys_gnu_openmpi
+    sbatch job_script.pr.bash
+
+You can also explore the utility in
+`utils/matrix <https://github.com/MPAS-Dev/compass/tree/main/utils/matrix>`_ to
+test on several compilers automatically.
+
+Testing MALI with Albany
```

Comment:
Maybe move this section after the next one so the two ocean flavors are together.

Reply:
Moved.